packages/core/src/shareback/cda-uploader.ts

```diff
@@ -0,0 +1,71 @@
+import { Organization } from "@medplum/fhirtypes";
+import { createUploadMetadataFilePath } from "../domain/document/upload";
+import { MetriportError } from "../util/error/metriport-error";
+import { createAndUploadCDAMetadataFile } from "./create-and-upload-extrinsic-object";
+import { createUploadFilePath } from "../domain/document/upload";
+import { uuidv7 } from "../util/uuid-v7";
+import { S3Utils } from "../external/aws/s3";
+import { out } from "../util/log";
+
+const MAXIMUM_FILE_SIZE = 50_000_000; // 50 MB
+const { log } = out(`Core CDA Uploader`);
+
+export async function cdaDocumentUploaderHandler({
+  cxId,
+  patientId,
+  cdaBundle,
+  destinationBucket,
+  region,
+  organization,
+}: {
+  cxId: string;
+  patientId: string;
+  cdaBundle: string;
+  destinationBucket: string;
+  region: string;
+  organization: Organization;
+}): Promise<void | { message: string; size: number }> {
+  const s3Utils = new S3Utils(region);
+
+  const docId = uuidv7();
+  const metadataFileName = createUploadMetadataFilePath(cxId, patientId, docId);
+  const destinationKey = createUploadFilePath(cxId, patientId, docId);
+
+  // Make a copy of the file to the general medical documents bucket
+  try {
+    await s3Utils.uploadFile(destinationBucket, `${destinationKey}.xml`, Buffer.from(cdaBundle));
+    log(`Successfully copied the uploaded file to ${destinationBucket} with key ${destinationKey}`);
+  } catch (error) {
+    const message = "Error copying the uploaded file to medical documents bucket";
+    log(`${message}: ${error}`);
+    throw new MetriportError(message, error, { destinationBucket, destinationKey });
+  }
+
+  try {
+    const { size, eTag } = await s3Utils.getFileInfoFromS3(destinationKey, destinationBucket);
+    const stringSize = size ? size.toString() : "";
+    const hash = eTag ? eTag : "";
```

Comment:
Can we use `undefined` when we don't have a value, please? [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#cff793e9f7f5445fb075c82490040bc1)

Reply:
Idk if @jonahkaye had something in mind with the hash, but the code simply passes it to both of these functions:
- createAndUploadDocumentdMetadataFile
- createAndUploadCDAMetadataFile
and neither of them do anything with it. 

I'll remove it for now, but lmk if there was a purpose to them

Reply:
Yes the hash was removed from the Metadata file since it was unnesssecary. Its not needed. Good catch!